test/functional/test-shell.md

```diff
@@ -0,0 +1,187 @@
+Test Shell for Interactive Environments
+=========================================
+
+This document describes the usage of the `TestShell` submodule in the functional
```

Comment:
nit: "describes how to use" might be friendlier.

Reply:
Fixed.